File: app/src/main/java/dev/ragnarok/fenrir/adapter/vkdatabase/SchoolClassesAdapter.java
package dev.ragnarok.fenrir.adapter.vkdatabase;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import dev.ragnarok.fenrir.R;
import dev.ragnarok.fenrir.model.database.SchoolClazz;

public class SchoolClassesAdapter extends RecyclerView.Adapter<SchoolClassesAdapter.Holder> {

    private final Context mContext;
    private final List<SchoolClazz> mData;
    private Listener mListener;

    public SchoolClassesAdapter(Context mContext, List<SchoolClazz> mData) {
        this.mContext = mContext;
        this.mData = mData;
    }

    @NonNull
    @Override
    public Holder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new Holder(LayoutInflater.from(mContext).inflate(R.layout.item_country, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull Holder holder, int position) {
        SchoolClazz schoolClazzDto = mData.get(position);
        holder.name.setText(schoolClazzDto.getTitle());

        holder.itemView.setOnClickListener(v -> {
            if (mListener != null) {
                mListener.onClick(schoolClazzDto);
            }
        });
    }

    @Override
    public int getItemCount() {
        return mData.size();
    }

    public void setListener(Listener listener) {
        mListener = listener;
    }

    public interface Listener {
        void onClick(SchoolClazz schoolClazz);
    }

    public static class Holder extends RecyclerView.ViewHolder {

        final TextView name;

        public Holder(View itemView) {
            super(itemView);
            name = itemView.findViewById(R.id.name);
        }
    }
}

File: app/src/main/java/dev/ragnarok/fenrir/adapter/vkdatabase/SchoolClassesAdapter.kt
package dev.ragnarok.fenrir.adapter.vkdatabase

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import dev.ragnarok.fenrir.R
import dev.ragnarok.fenrir.model.database.SchoolClazz

class SchoolClassesAdapter(private val mContext: Context, private val mData: List<SchoolClazz>) :
    RecyclerView.Adapter<SchoolClassesAdapter.Holder>() {
    private var mListener: Listener? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Holder {
        return Holder(
            LayoutInflater.from(
                mContext
            ).inflate(R.layout.item_country, parent, false)
        )
    }

    override fun onBindViewHolder(holder: Holder, position: Int) {
        val schoolClazzDto = mData[position]
        holder.name.text = schoolClazzDto.title
        holder.itemView.setOnClickListener {
            mListener?.onClick(schoolClazzDto)
        }
    }

    override fun getItemCount(): Int {
        return mData.size
    }

    fun setListener(listener: Listener?) {
        mListener = listener
    }

    interface Listener {
        fun onClick(schoolClazz: SchoolClazz)
    }

    class Holder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val name: TextView = itemView.findViewById(R.id.name)
    }
}